### PR TITLE
Refactor PKO

### PR DESCRIPTION
**Pull request by janboll**
Adding predeploy step
Updating deploy step, changing to three different secrets

### CODE REVIEW

File: pko/package/manifest.yaml [hunk 1]
```diff
@@ -43,11 +44,11 @@ spec:
             imageTag:
               type: string
               description: tag of the image containing unleash
-              default: UNLEASH_IMAGE_TAG
+              default: latest
```
Comment (geoberle) [inline pko/package/manifest.yaml:47]: UNLEASH_IMAGE_TAG was meant to be a default to keep the PKO package and the unleash release in sync. it is a variable replaced by the 7digit sha during the build of the package.

what is the reason to replace it with latest?

Comment (janboll) [reply pko/package/manifest.yaml]: Was for local testing only

File: pko/package/manifest.yaml [hunk 2]
```diff
@@ -47,7 +48,7 @@ spec:
             replicas:
               description: number of replicas for the unleash deployment
               type: integer
-              default: 3
+              default: 1
```
Comment (hemslo) [inline pko/package/manifest.yaml:51]: 1 replica may trigger DVO alert

Comment (hemslo) [reply pko/package/manifest.yaml:51]: btw `resources` section, will `limits.cpu` always default to `100m` even not specified?

Comment (janboll) [reply pko/package/manifest.yaml:51]: Right, 1 is to conservative here

**Review by geoberle — APPROVED** [review]

**Review by hemslo — APPROVED** [review]

**Pull request merged**